app/src/main/java/com/android/streetworkapp/model/moderation/PerspectiveAPIRepository.kt

```diff
@@ -0,0 +1,133 @@
+package com.android.streetworkapp.model.moderation
+
+import android.util.Log
+import com.android.sample.BuildConfig
+import java.net.HttpURLConnection.HTTP_BAD_REQUEST
+import java.net.HttpURLConnection.HTTP_OK
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.encodeToJsonElement
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+
+class PerspectiveAPIRepository(private val client: OkHttpClient) : TextModerationRepository {
+  private val DEBUG_PREFIX = "PerspectiveAPIRepository:"
+
+  override fun evaluateText(content: String, thresholds: Map<TextModerationTags, Double>): Boolean {
+    if (content.isEmpty()) return false
+
+    when (val result = this.getTextAnnotations(content)) {
+      is TextEvaluationResult.Error -> {
+        Log.d(this.DEBUG_PREFIX, result.errorType.errorMessage)
+        return false
+      }
+      is TextEvaluationResult.Success -> {
+        for (resultAnnotation in result.annotations) {
+          thresholds[resultAnnotation.tag]?.let { thresholdTagProbability ->
+            if (resultAnnotation.probability > thresholdTagProbability)
+                return false // Text over one of the thresholds
+          }
+              ?: run {
+                return false // invalid tag
+              }
+        }
+
+        return true // all the tags under the thresholds :)
+      }
+    }
+  }
+
+  /**
+   * Gets the tags and their probabilities for param content
+   *
+   * @param content Text to be analyzed
+   * @return TextEvaluationResult.Success if the API could process the content,
+   *   TextEvaluationResult.Error if an error was encountered
+   */
+  private fun getTextAnnotations(content: String): TextEvaluationResult {
+    // I don't perform any checks on the content, I'll let the api handle the error checking
+
+    val requestMediaType = "application/json; charset=utf-8".toMediaType()
+    val requestBody = this.formatPostRequestBody(content).toRequestBody(requestMediaType)
+
+    // note: someone could get the key by sniffing the packets, but since we can't have a backend
+    // here we are :)
+    val request =
+        Request.Builder()
```

Comment:
Suggestion: There is a "cleaner" way to build a URL. Here is for instance what we used for `Nominatim` : 

```
val url =
        HttpUrl.Builder()
            .scheme("https")
            .host("nominatim.openstreetmap.org")
            .addPathSegment("lookup")
            .addQueryParameter("osm_ids", "W$locationId")
            .addQueryParameter("format", "json")
            .build()
```

(Note: this is non critical, if it creates any problems with this specific query you can ignore this suggestion)

Reply:
Didn't know about it, it's done